api/src/main/java/org/pmiops/workbench/rdr/RdrTaskQueue.java

```diff
@@ -68,27 +71,20 @@ public void groupIdsAndPushTask(List<Long> idList, String taskUri) throws IOExce
     }
   }
 
-  private static void createAndPushTask(List<Long> ids, String queuePath, String taskUri)
+  private void createAndPushTask(List<Long> ids, String queuePath, String taskUri)
       throws IOException {
-    List<String> idsAsString = ids.stream().map(id -> id.toString()).collect(Collectors.toList());
     Gson gson = new Gson();
     String jsonIds = gson.toJson(ids);
-    try (CloudTasksClient client = CloudTasksClient.create()) {
-      AppEngineHttpRequest req =
-          AppEngineHttpRequest.newBuilder()
-              .setRelativeUri(taskUri)
-              .setBody(ByteString.copyFromUtf8(jsonIds))
-              .setHttpMethod(HttpMethod.POST)
-              .putHeaders("Content-type", "application/json")
-              .build();
-      client.createTask(queuePath, Task.newBuilder().setAppEngineHttpRequest(req).build());
-    } catch (IOException ex) {
-      log.severe(
-          String.format(
-              "Error while creating task to push to queue for IDS %s and path %s. "
-                  + "Re-throwing error",
-              idsAsString, taskUri));
-      throw ex;
-    }
+    CloudTasksClient client = cloudTasksClientProvider.get();
+    client.createTask(
```

Comment:
minor cleanup. Try/catch was for creation of the cloud tasks client, which now happens in Spring